### ISSUE TITLE
OSX Yosemite Compatibility

### ISSUE THREAD

**Issue by srenne7** [greneholt/youtube5#59]
Currently this app doesn't seem to be compatible with OS X Yosemite, as youtube videos display normally:(

**Comment by Anthracity** [comment]
I have the same problem in the Yosemite Public Beta. No errors or anything – YouTube clips just play back using the YouTube Player instead of YouTube5. Tried reinstalling it after the Yosemite update, but that didn't help.

**Comment by dkratochvil** [comment]
At some point Youtube started recognizing Safari 8's user agent and forcing it to use their html5 player. It also does this for Chrome and Firefox. However, you can change the user agent that Safari 8 uses to the older 7.1 version that Mavericks uses with a terminal command, and restore functionality. Here it is:

First Quit Safari. Then open Terminal and paste this command, and press enter:

defaults write com.apple.Safari CustomUserAgent "\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_0) AppleWebKit/538.46 (KHTML, like Gecko) Version/7.1 Safari/537.85.7\""

Open Safari and you're done.

To undo this change, close Safari and paste this command into terminal:

defaults delete com.apple.Safari CustomUserAgent

Sometimes a restart may be required to get these changes to stick, not sure why, could be a cache thing.

**Comment by mikesolis** [comment]
That user agent change definitely works D, nice idea. Not sure if it's just me but unfortunately the video persists on the screen when you go back, like so:

![screen shot 2014-08-03 at 13 45 33](https://cloud.githubusercontent.com/assets/7841357/3791301/5102a0d4-1b36-11e4-9da0-22bae4db7758.png)

Additionally, I would guess that this user agent change will affect Netflix's use of their new HTML5 player that is only supported by Safari 8. I don't use Netflix but I figured I'd put that out there. Either way it's better than nothing!

**Comment by voidref** [comment]
Today it started working again, w00t!

**Comment by ghost** [comment]
Just go to the Develop menu in Safari, change the user agent to Safari 7 or 6. Then visit the http://youtube.com/html5 page and select to use the Default Player.

As long as you browse YouTube with Safari's user agent switched to Safari 6 or 7 YouTube5 will work again, but it's a bit of a drag having to switch the user agent manually every time I visit YouTube…

**Comment by mvarie** [comment]
@arglborps works beautifully, thanks.

**Comment by ghost** [comment]
I haven't installed Yosemite yet, is the user agent fix still necessary? @voidref's comment seems to indicate that it isn't the case.

**Comment by voidref** [comment]
The user agent fix became necessary shortly after it started working for me.

What happens now is that it will use Google's html 5 player, which isn't as good at the YouTube 5 one for buffering or performance. (but will do 1080p, and show you ads and annotations).

**Comment by sxflynn** [comment]
Is it possible for Youtube5 to replace the HTML5 player that Youtube uses?

**Comment by greneholt** [comment]
Yes, it is possible, it just requires a lot of changes to how YouTube5 works.

**Comment by ghost** [comment]
Marc Hoyois has found a way to do this. See this discussion: https://github.com/hoyois/clicktoplugin/issues/78

**Comment by greneholt** [comment]
Very interesting, that's not the route I'm taking, but it's good to know how it was done. My goal is to still have youtube5 work with flash uninstalled.

**Comment by ghost** [comment]
CTP definitely works with Flash uninstalled for me. With both Yosemite and Mavericks.

**Comment by stephenreay** [comment]
An alternative I've discovered (since even the UA change stopped working for me today) - load the YT video in an iframe ( I just used a simple html file ).

Possibly The extension could detect a native YT player and replace it with an iframe pointing back to the embed page for that video?

**Comment by jonathanlang** [comment]
I recommend quietube.com for a quick way to get an iFrame of a Youtube-Video - and the page is nice and clean, too! :)
